pkg/controller/kogitoinfra/mongodb.go

```diff
@@ -0,0 +1,308 @@
+	return nil
+}
+
+func updateMongoDBEnvVarsInStatus(cli *client.Client, mongoDBInstance *mongodb.MongoDB, instance *v1alpha1.KogitoInfra, scheme *runtime.Scheme) error {
+	log.Debugf("going to Update MongoDB env properties in kogito infra instance status")
+	envVars, err := getMongoDBSecretEnvVars(cli, mongoDBInstance, instance, scheme)
+	if err != nil {
+		return err
+	}
+	instance.Status.Env = envVars
+	log.Debugf("Following env properties are set infra status : %s", envVars)
+	return nil
+}
+
+func loadDeployedMongoDBInstance(cli *client.Client, instanceName string, namespace string) (*mongodb.MongoDB, error) {
+	log.Debug("fetching deployed kogito mongoDB instance")
+	mongoDBInstance := &mongodb.MongoDB{}
+	if exits, err := kubernetes.ResourceC(cli).FetchWithKey(types.NamespacedName{Name: instanceName, Namespace: namespace}, mongoDBInstance); err != nil {
+		log.Error("Error occurs while fetching kogito mongoDB instance")
+		return nil, err
+	} else if !exits {
+		log.Debug("Kogito mongoDB instance is not exists")
+		return nil, nil
+	} else {
+		log.Debug("Kogito mongoDB instance found")
+		return mongoDBInstance, nil
+	}
+}
+
+func createNewMongoDBInstance(cli *client.Client, name string, namespace string, instance *v1alpha1.KogitoInfra, scheme *runtime.Scheme) (*mongodb.MongoDB, error) {
+	log.Debug("Going to create kogito mongoDB instance")
+
+	// Create default secret
+	secret := infrastructure.GetMongoDBDefaultSecret(namespace)
+	if err := kubernetes.ResourceC(cli).Create(secret); err != nil {
+		log.Error("Error occurs while creating secret for kogito mongoDB instance")
+		return nil, err
+	}
+
+	// Create mongoDB instance
+	mongoDBRes := infrastructure.GetMongoDBStub(cli.IsOpenshift(), namespace, infrastructure.MongoDBInstanceName, nil)
+	if err := controllerutil.SetOwnerReference(instance, mongoDBRes, scheme); err != nil {
+		return nil, err
+	}
+	if err := kubernetes.ResourceC(cli).Create(mongoDBRes); err != nil {
+		log.Error("Error occurs while creating kogito mongoDB instance")
+		return nil, err
+	}
+	log.Debug("Kogito mongoDB instance created successfully")
+	return mongoDBRes, nil
+}
+
+func loadCustomKogitoMongoDBSecret(cli *client.Client, namespace string) (*corev1.Secret, error) {
+	log.Debugf("Fetching %s ", mongoDBSecretName)
+	secret := &corev1.Secret{}
+	if exits, err := kubernetes.ResourceC(cli).FetchWithKey(types.NamespacedName{Name: mongoDBSecretName, Namespace: namespace}, secret); err != nil {
+		log.Errorf("Error occurs while fetching %s", mongoDBSecretName)
+		return nil, err
+	} else if !exits {
+		log.Errorf("%s not found", mongoDBSecretName)
+		return nil, nil
+	} else {
+		log.Debugf("%s successfully fetched", mongoDBSecretName)
+		return secret, nil
+	}
+}
+
+func createCustomKogitoMongoDBSecret(cli *client.Client, namespace string, mongoDBInstance *mongodb.MongoDB, instance *v1alpha1.KogitoInfra, scheme *runtime.Scheme) (*corev1.Secret, error) {
```

Comment:
Could you please add a reference to the MongoDB operator documentation that mentions this use case?